src/Dispatcher/Result.php

```diff
@@ -0,0 +1,185 @@
+<?php
+declare(strict_types=1);
+
+namespace FastRoute\Dispatcher;
+
+use FastRoute\RouteInterface;
+use RuntimeException;
+
+/**
+ * Result Object
+ */
+class Result implements ResultInterface
+{
+    public const NOT_FOUND = 0;
+    public const FOUND = 1;
+    public const METHOD_NOT_ALLOWED = 2;
+
+    protected bool $matched = false;
+
+    protected ?RouteInterface $route;
+
+    /** @var mixed[] */
+    protected array $result = [];
+
+    protected int $status = self::NOT_FOUND;
+
+    /** @var mixed */
+    protected $handler;
+
+    /** @var mixed[] */
+    protected array $args = [];
+
+    /** @var string[] */
+    protected array $allowedMethods = [];
```

Comment:
This object is a DTO and I believe we can simplify the API by using public properties.
I'd also change `$args` to `$variables` to avoid abbreviations and making things more explicit.

```suggestion
    public int $status = self::NOT_FOUND;

    /** @var mixed */
    public $handler;

    /** @var mixed[] */
    public array $variables = [];

    /** @var string[] */
    public array $allowedMethods = [];
```

Reply:
I disagree about using properties. They'll be also writeable from a context from where they should not and if somebody wants to alter the behavior they can't without changing the logic with that simple.